client/Bootstrap/Navigation/useRouteConfig/useRouteConfig.hook.ts

```diff
@@ -67,29 +88,43 @@ const mergeReduce = (acc, cur) =>
   });
 
 const useRouteConfig = (input: PageConfig): RouterConfig => {
-  const leaves: Array<Leaf> = [];
-  const output = Object.values(input)
-    .map((value) => {
-      const page = value as Page;
-      const contexts = page.contexts;
-
-      return contexts.reduce(
-        (acc, context) => contextReducer(acc, context, page, leaves),
-        {
-          links: [],
-          routes: [],
-          meta: {},
-        }
-      );
-    })
-    .reduce(mergeReduce, {} as RouterConfig);
-
-  return leaves.reduce((acc, leaf) => {
-    if (acc.meta[leaf.parent]) {
-      acc.meta[leaf.parent].leaves.push(leaf.child);
-    }
-    return acc;
-  }, output);
+  const { t: translate } = useTranslation();
+  const { featureFlags, isComplete } = useConfigFeatureFlag();
+
+  if (isComplete) {
+    const leaves: Array<Leaf> = [];
+    const output = Object.values(input)
+      .map((value) => {
+        const page = value as Page;
+        const contexts = page.contexts;
+
+        return contexts.reduce(
+          (acc, context) =>
+            contextReducer(acc, context, page, leaves, featureFlags, translate),
+          {
+            links: [],
+            routes: [],
+            meta: {},
+            isComplete,
+          }
+        );
+      })
+      .reduce(mergeReduce, {} as RouterConfig);
+
+    return leaves.reduce((acc, leaf) => {
+      if (acc.meta[leaf.parent]) {
```

Comment:
Same principle re: reduce - it's more of a map/foreach if we can't do it inside the previous reduce.